src/response/error.rs

```diff
@@ -4,108 +4,124 @@ use tonic::codegen::http;
 
 use crate::auth::AuthError;
 
+#[derive(Debug)]
+pub enum MomentoErrorCode {
+    /// Invalid argument passed to Momento client
+    InvalidArgumentError,
+    /// Service returned an unknown response
+    UnknownServiceError,
+    /// Resource with specified name already exists
+    AlreadyExistsError,
+    /// Cache with specified name doesn't exist
+    NotFoundError,
+    /// An unexpected error occurred while trying to fulfill the request
+    InternalServerError,
+    /// Insufficient permissions to perform operation
+    PermissionError,
+    /// Invalid authentication credentials to connect to service
+    AuthenticationError,
+    /// Request was cancelled by the server
+    CancelledError,
+    /// Request rate, bandwidth, or object size exceeded the limits for the account
+    LimitExceededError,
+    /// Request was invalid
+    BadRequestError,
+    /// Client's configured timeout was exceeded
+    TimeoutError,
+    /// Server was unable to handle the request
+    ServerUnavailable,
+    /// A client resource (most likely memory) was exhausted
+    ClientResourceExhausted,
+    /// System is not in a state required for the operation's execution
+    FailedPreconditionError,
+    /// Unknown error has occurred
+    UnknownError,
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("{details}")]
+pub struct MomentoGrpcErrorDetails {
+    pub code: tonic::Code,
+    pub details: Cow<'static, str>,
+    pub metadata: Option<tonic::metadata::MetadataMap>,
```

Comment:
It looks to me like the tonic `Status` struct always has a Metadata on it, so I think this should not be `Option`?